jccli/cli/__init__.py

```diff
@@ -22,18 +23,28 @@
 @click.group()
 @click.option('--key', "-k", type=str, help='Jumpcloud API key (can also use environmental variable: JC_API_KEY)',
               envvar='JC_API_KEY')
-@click.option('--key-file', '-K', type=str, help='Path to text file containing Jumpcloud API key')
+@click.option('--profile', '-p', help='A user profile, as specified in the config file', default='DEFAULT',
+              show_default=True)
 @click_log.simple_verbosity_option(LOGGER)
 @click.version_option(version=__version__)
 @click.pass_context
-def cli(ctx, key, key_file):
+def cli(ctx, key, profile):
     """
     Run jccli.
     """
-    if key and key_file:
-        sys.exit("Please provide a key *or* a key file, not both")
-    if key_file and not key:
-        key = open(key_file).read()
+    try:
+        config = load_config(profile)
+    except KeyError:
+        sys.exit("no profile found named: %s" % (profile,))
```

Comment:
It doesn't really seem like it needs to necessarily create a default config file if one does not exist (i.e. if there is no config file and it has what it needs from defaults/command line options/env variables then maybe nice not to add a file that has no functional effect to the user's home directory.

Reply:
It doesn't make sense with the current state of the CLI. But in the near future, I want to add output formatting settings (changeable in config profile and via command line options). I think it might be nice to auto-populate a config file with the default formatting settings? I just feel like it's easier to edit a config file when you have an example to build on.

E.g. the (auto-generated) default might be:

```
[DEFAULT]
format=json
indent=4
```

which would make it really intuitive for the user to change it to

```
[DEFAULT]
format=yaml
indent=2
```

Does that make sense? You think worth it or no?

Reply:
i don't think it's worth maintaining code to auto generate config file.

Reply:
Fair enough. I removed the config auto-generation.